filecoin-proofs/src/api/post_util.rs

```diff
@@ -100,9 +102,16 @@ pub fn generate_fallback_sector_challenges<Tree: 'static + MerkleTreeTrait>(
             let mut challenges = Vec::new();
 
             for n in 0..post_config.challenge_count {
-                let challenge_index = ((partition_index * post_config.sector_count + i)
-                    * post_config.challenge_count
-                    + n) as u64;
+                let sector_index = match post_config.typ {
+                    PoStType::Window => partition_index * num_sectors_per_chunk + i,
+                    PoStType::Winning => partition_index * post_config.sector_count + i,
```

Comment:
Maybe in the future (not in this PR) we should just put an assertion that if post-type is Winning-PoSt, that `post_config.sector_count == pub_sectors.len() == 1`, so `sector_index = 0` can be hardcoded.